### PR TITLE
Added loading state and expose with prop

### PR DESCRIPTION
**Pull request by mikefrancis**
Implements #46 

I have prettier installed in my editor and it seems to have reformatted the whole file 🤦‍♂ 

Is there a config you use for this repo so I can format it back? It's going be time consuming to cherry pick the changes I've made.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -22,3 +22,5 @@ yarn-error.log*
 
 .rpt*
 coverage
+
+yarn.lock
```
Comment (mikefrancis) [inline .gitignore]: This should prevent the `yarn.lock` appearing in future PRs 👍

**Comment by armand1m** [comment]
@mikefrancis you can make the changes you want and just use VSCode "Save without Formatting" : )

**Comment by mikefrancis** [comment]
@armand1m Thanks - I am aware of that, I was just working on this in my spare time so when I'm switching to other projects and back again I am unable to "Undo" and then "Save without formatting" retrospectively.

**Comment by armand1m** [comment]
@mikefrancis would be nice then for you to add your .prettierrc config to the project then =) so it'll become the default one for this

**Comment by mikefrancis** [comment]
@armand1m Ok, I've tried to match the prettier config to what it was before (single quotes) and have run.

**Pull request merged**